Source/WebInspectorUI/UserInterface/Models/Resource.js

```diff
@@ -884,6 +884,14 @@ WI.Resource = class Resource extends WI.SourceCode
             console.assert(scriptForTarget);
             if (scriptForTarget)
                 return scriptForTarget.requestContentFromBackend();
+        } else if (this.type === WI.Resource.Type.StyleSheet) {
+            return new Promise((resolve, reject) => {
+                WI.cssManager.lookupStyleSheetForResource(this, (cssStyleSheet) => {
+                    this._target.CSSAgent.getStyleSheetText(cssStyleSheet.id)
+                        .then((text) => resolve(text))
+                        .catch((error) => reject(error));
+                });
+            });
```

Comment:
Is there a prettier way to write this? (Perhaps with `async` and `await`, but I couldn't see how.)

Reply:
Also,  this was done to make fetching CSS text use `CSS.getStyleSheetText` instead of the normal resource-fetching routine to ensure CSSOM-modified contents can show up in Sources without needing the user to visit the Elements tab first.